### ISSUE TITLE
[expect-expect]: Unable to allow methods in `assertFunctionNames`

### ISSUE THREAD

**Issue by randycoulman** [jest-community/eslint-plugin-jest#175]
In testing the `expect-expect` rule on one of my projects, I found that it was reporting a number of tests that didn't have expect calls.  In these tests, I was using `td.verify` from [testdouble.js](https://github.com/testdouble/testdouble.js/) which verifies that a particular test double was called.

I attempted to use the `assertFunctionNames` option to configure `td.verify` as a known expect function, but was unable to get it to work.  The tests in questions were still reported as violating this rule.

I tried both `assertFunctionNames: ["expect", "td.verify"]` and `assertFunctionNames: ["expect", "verify"]` and neither approach worked.

I was able to work around the issue by doing the following, but this is not the normal way to use testdouble.js and I'd prefer not to have to do it everywhere I use `td.verify`.

```js
const { verify } = td;

test("...", () => {
  // ...
  verify(someFunctionCall());
}
```

I think the more general problem here is that the rule is not able to match anything but bare functions; it can't handle methods on an object.

I don't know the AST syntax well enough to know how to fix this, but I thought I'd at least report it here in the hope that someone more familiar can take care of it.

**Comment by macklinu** [comment]
Thanks for reporting this issue with clear steps and a use case. It'd be good to consider a call expression with an object and property (like `td.verify`). 👍 PRs welcome from anyone, will try to take a look at this this weekend as well. 😄

**Comment by macklinu** [comment]
I can push up a PR for this shortly.
